app/main.ts

```diff
@@ -24,52 +53,155 @@ async function start(): Promise<void> {
+    const contextMenu = Menu.buildFromTemplate([
+      {
+        label: `Show Gatsby Desktop`,
+        click: openMainWindow,
+      },
+      {
+        label: `Quit...`,
+        click: async (): Promise<void> => {
+          openMainWindow()
+          const { response } = await dialog.showMessageBox({
+            message: `Quit Gatsby Desktop?`,
+            detail: `This will stop all running sites`,
+            buttons: [`Cancel`, `Quit`],
+            defaultId: 1,
+            type: `question`,
+          })
+
+          if (response === 1) {
+            app.quit()
+          }
+        },
+      },
+    ])
+    tray.setContextMenu(contextMenu)
 
-  ipcMain.on(`watch-sites`, (event) => {
     watchSites((sites) => {
-      console.log(`Got sites`, sites)
-      event.sender.send(`sites-updated`, sites)
+      siteList = sites.map((site) => {
+        const launcher = site.hash && siteLaunchers.get(site.hash)
+        if (!launcher) {
+          return {
+            site,
+          }
+        }
+        const { logs, rawLogs, status } = launcher
+        return {
+          site,
+          status: { startedInDesktop: true, logs, rawLogs, status },
+        }
+      })
+      if (mainWindow && !mainWindow.isDestroyed()) {
+        mainWindow?.webContents?.send(`sites-updated`, siteList)
+      }
     })
+
+    // If we're not running develop we can preload the start page
+
+    if (!process.env.GATSBY_DEVELOP_URL) {
+      mainWindow.loadURL(makeUrl(`sites`))
+      mainWindow.show()
+    }
   })
 
-  ipcMain.on(`unwatch-sites`, () => {
-    stopWatching()
+  ipcMain.on(`open-main`, async (event, payload?: string) => {
+    if (!mainWindow || mainWindow.isDestroyed()) {
+      mainWindow = makeWindow()
+    }
+    const url = makeUrl(payload ? `sites/${payload}` : `sites`)
+    if (mainWindow.webContents.getURL() !== url) {
+      mainWindow.loadURL(url)
+    }
+    mainWindow.show()
   })
 
+  ipcMain.on(
+    `launch-site`,
+    async (event, { root, hash, clean }): Promise<number | undefined> => {
+      if (!root || !hash) {
+        console.error(`Invalid launch params`, { root, hash })
+        return undefined
+      }
+      let launcher = siteLaunchers.get(hash)
+      if (!launcher) {
+        launcher = new SiteLauncher(root, hash)
+        launcher.setListener((message: Message) => {
+          if (mainWindow && !mainWindow.isDestroyed()) {
+            mainWindow.webContents?.send(`site-message`, hash, message)
+          }
+        })
+        siteLaunchers.set(hash, launcher)
+      }
+      return launcher.start(clean)
+    }
+  )
+
+  ipcMain.on(
+    `stop-site`,
+    async (event, { hash, pid }): Promise<void> => {
+      if (!hash) {
+        console.error(`Missing site hash`)
+        return
+      }
+      const launcher = siteLaunchers.get(hash)
+      if (launcher) {
+        launcher.stop()
+        return
+      }
+      if (pid) {
+        console.error(`Site not found`)
+      }
+      process.kill(pid)
```

Comment:
Would you still call `process.kill` if pid is falsy? Potentially even `0` (which you can't kill).